`TwoPowerProfile`

ddudt opened the pull request:
New class for profiles of the form $f(\rho) = a(1-\rho^b)^c$ where $a$, $b$, $c$ are all floats. This is equivalent to the `"two_power"` profile input format in VMEC.

f0uriest commented on desc/profiles.py:
```diff
@@ -687,6 +692,88 @@ def from_values(cls, x, y, order=6, rcond=None, w=None, sym="auto", name=""):
+        r = grid.nodes[:, 0]
+        if dr == 0:
+            f = a * (1 - r**b) ** c
+        # df/dr = inf at rho = 1 for all dr > 0
```
isnt this only true if b,c are floats <dr?

what is the use case for b,c being floats rather than ints?

my worry is also that if b<1 than the derivative blows up at r=0 as well.

ddudt replied:
True, depending on the parameter values it may or may not blow up at $\rho=0$ and/or $\rho=1$. 

My personal use case is to use an integer `b` = 1 or 2, and `c` as a float. But if these numbers are optimizable params they all need to be floats. 

Options:

1. Add a check to make sure `b >= 1`. But there isn't much we can do about `c`. 
2. We don't need to merge this into the master branch. But I coded it up anyways so I figured I would share it.

f0uriest replied:
My question is, do they all need to be optimizable? We could just make `b`,`c` constant integers and leave `a` optimizable?

Along a similar vein, VMEC has a bunch of other profiles types that might be worth looking at: https://github.com/jonathanschilling/educational_VMEC/blob/master/src/profile_functions.f

I have some VMEC input files that use the atan and two-lorentz profiles that would be good to try in DESC.

dpanici commented on desc/profiles.py:
```diff
@@ -687,6 +692,89 @@ def from_values(cls, x, y, order=6, rcond=None, w=None, sym="auto", name=""):
+        r = grid.nodes[:, 0]
+        if dr == 0:
+            f = a * (1 - r**b) ** c
+        # df/dr = inf at rho = 0 if b < 1
```
is it even physical then if b < 1?

dpanici reviewed (changes requested):
- add higher derivs
- add to docstring that the params are only good for certain parameter ranges
- add warning in build to warn if the parameters make things be divergent or nonphysical
- check if it overlaps with a VMEC one and if so, then we can say it is similar to that

ddudt commented:
> * add higher derivs
> * add to docstring that the params are only good for certain parameter ranges
> * add warning in build to warn if the parameters make things be divergent or nonphysical
> * check if it overlaps with a VMEC one and if so, then we can say it is similar to that

- It turns out this is the same as the `"two_power"` profiles in VMEC, so I changed the name accordingly. 
- The `MTanhProfile` is also only implemented up to derivative order `dr=2`. Do we actually need higher order derivatives? 
- I added documentation and warnings for bad parameter values at initialization.

dpanici commented on desc/profiles.py:
```diff
@@ -687,6 +693,102 @@ def from_values(cls, x, y, order=6, rcond=None, w=None, sym="auto", name=""):
+        errorif(
+            self._params.size != 3, ValueError, "params must be an array of size 3."
+        )
+        warnif(
```
add why this is occuring (a[1]<1) in the warning string, I know this in the docstring but the warning is what will pop up first when this is encountered

dpanici commented on desc/profiles.py:
```diff
@@ -687,6 +693,102 @@ def from_values(cls, x, y, order=6, rcond=None, w=None, sym="auto", name=""):
+                * self.compute(grid, params=[a * b * c, b, c - 2])
+            )
+        else:
+            raise NotImplementedError("dr > 2 not implemented for TwoPowerProfile!")
```
add a test for this (is one line)

dpanici reviewed (changes requested):
see comments
